apps/finance/app/src/components/useFilteredTransfers.js

```diff
@@ -0,0 +1,106 @@
+import { useState, useCallback, useMemo } from 'react'
+import { endOfDay, isWithinInterval, startOfDay } from 'date-fns'
+import { TRANSFER_TYPES, Incoming, Outgoing } from '../transfer-types'
+import { addressesEqual } from '../lib/web3-utils'
+
+const UNSELECTED_TOKEN_FILTER = -1
+const UNSELECTED_TRANSFER_TYPE_FILTER = -1
+const UNSELECTED_DATE_RANGE_FILTER = { start: null, end: null }
+
+function useFilteredTransfers({ transactions, tokens }) {
+  const [page, setPage] = useState(0)
```

Comment:
Just wondering, do we control the page like this in Agent too?

It's kind of related... but it also feels like it doesn't really belong here since this hook should just be controlling the filtering.

Reply:
Yes, it's done this way as well. I took this decision because we have to reset the page if any of the filtering options change; originally this was done on the component holding the DataView, but as we had to handle each case, it felt like it was leaking filtering logic into the DataView component.

Reply:
Yes, the alternative would be running it as an effect in the component using this hook, such that pages is only dependent on the transaction list changing.